### PR TITLE
Ballot checking integration

### PR DESCRIPTION
**Pull request by av-mads**
I dont know where these changes should go so I have targeted `main` for now. The commit history, however, suggests these changes haven't been merged for a while, if ever.

The changes I have made is updating the client-service alongside a few UI changes where we needed to display the tracking and pairing code. 

The `pages/check-your-pairing-codes/check-your-pairing-codes.page.ts-file` is missing an error handling function which directs the user back to the start of the voter flow if a `TimeoutError` happens (i have left a comment). 

@mattbeattie If you have any questions tag me in the PR or write me on Slack. I'm also open to a pairing session if it is needed.

### CODE REVIEW

File: src/environments/environment.ts [hunk 1]
```diff
@@ -4,8 +4,7 @@
 
 export const environment = {
   production: false,
-  // note: this is a temporary value to satisfy function signatures; it does not suggest that a server should be running on port 3000
-  url: 'http://localhost:3000/',
+  url: 'http://avx:3000/dbb/oset/api',
```
Comment (mattbeattie) [inline src/environments/environment.ts:7]: Can you revert this change? We shouldn't have anything with AV in it.

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,3 +1,3 @@
 <ion-app>
-  <ion-router-outlet></ion-router-outlet>
+  <ion-router-outlet swipeGesture="false" animated="false"></ion-router-outlet>
```
Comment (mattbeattie) [inline src/app/app.component.html:2]: Can you explain why this change was made?

File: src/app/api/artifacts/dr-sample-cvr.ts [hunk 3]
```diff
@@ -1,149 +1,21 @@
-export const DR_SAMPLE_CVR = `<?xml version="1.0" encoding="UTF-8"?>
-<CastVoteRecordReport xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://itl.nist.gov/ns/voting/1500-103/v1 ../../../../../CastVoteRecords/NIST_V0_cast_vote_records.xsd"
-    xmlns="http://itl.nist.gov/ns/voting/1500-103/v1">
+export const DR_SAMPLE_CVR = `
```
Comment (mattbeattie) [inline src/app/api/artifacts/dr-sample-cvr.ts:1]: What was the reason for this change? Can it be reverted?

File: package-lock.json [hunk 4]
```diff
@@ -879,9 +879,8 @@
     },
     "node_modules/@babel/helper-plugin-utils": {
       "version": "7.16.7",
-      "resolved": "https://registry.npmjs.org/@babel/helper-plugin-utils/-/helper-plugin-utils-7.16.7.tgz",
```
Comment (mattbeattie) [inline package-lock.json:882]: No dependencies were added/removed. Please revert all changes to this file.

File: config.xml [hunk 5]
```diff
@@ -18,6 +18,7 @@
     <preference name="SplashShowOnlyFirstTime" value="false" />
     <preference name="SplashScreen" value="screen" />
     <preference name="SplashScreenDelay" value="3000" />
+    <preference name="AllowBackForwardNavigationGestures" value="false" />
```
Comment (mattbeattie) [inline config.xml:21]: Similar to the other question, why was this changed?

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -1,9 +1,9 @@
 name: mvp-test-harness
 on:
   push:
-    branches: [main]
+    branches: [main, av-floating]
```
Comment (mattbeattie) [inline .github/workflows/ci.yml:4]: Can you revert all changes to this file?

**Review by mattbeattie — DISMISSED** [review]
Hey Mads - a few questions/requests below, let me know what you think.

Additionally, it looks like CI is failing for this PR; can you make the necessary updates so that everything is green?

Last, it appears there are a lot of commits in this PR. Can you enumerate the material changes here for posterity?

File: src/environments/environment.ts (hunk 1 above)
Comment (mattbeattie) [reply src/environments/environment.ts:7]: One caveat: this change _does_ make sense to keep in the forked code for the demo; however, it shouldn't be merged to the `main` branch of the repo.

File: config.xml (hunk 5 above)
Comment (AV-MadsChris) [reply config.xml:21]: I dont know, it seems this change was introduced by @shreedhanu08 when creating the new `CheckPairingCode`-screen.

File: src/app/api/artifacts/dr-sample-cvr.ts (hunk 3 above)
Comment (AV-MadsChris) [reply src/app/api/artifacts/dr-sample-cvr.ts:1]: We are using a new CVR determined by the new EDFs. We are using it for integration tests with the `Provisioner`. Is there an issue with changing the CVR?

File: src/app/app.component.html (hunk 2 above)
Comment (AV-MadsChris) [reply src/app/app.component.html:2]: Again, this change was introduced by @shreedhanu08

File: src/environments/environment.ts (hunk 1 above)
Comment (AV-MadsChris) [reply src/environments/environment.ts:7]: I will revert it

File: package-lock.json (hunk 4 above)
Comment (AV-MadsChris) [reply package-lock.json:882]: This was also introduces by @shreedhanu08 as far as I can see

**Comment by AV-MadsChris** [comment]
@mattbeattie I have fixed as much as I could. I have also fixed as many tests as I could, however, one is still failing but I think it should be fixed by you or @shreedhanu08 as I didn't write it and I'm not familiar with the test framework.

File: src/app/app.component.html (hunk 2 above)
Comment (mattbeattie) [reply src/app/app.component.html:2]: Interesting, where was it introduced? It not in the main branch at this time:

https://github.com/trustthevote/MVP-App-TestHarness/blob/main/src/app/app.component.html

File: src/app/api/artifacts/dr-sample-cvr.ts (hunk 3 above)
Comment (mattbeattie) [reply src/app/api/artifacts/dr-sample-cvr.ts:1]: Great question - @awoflaherty what do you think?

File: package-lock.json (hunk 4 above)
Comment (mattbeattie) [reply package-lock.json:882]: Right, but they don't exist in the main branch. So as the PR owner, _you_ are the driving contributor for _all_ of the changes in this PR - not just the ones that you made.

You'll need to either remove any changes that you didn't make, or take ownership of the changes that you're looking to get merged.

File: config.xml (hunk 5 above)
Comment (mattbeattie) [reply config.xml:21]: Same response as above - as the PR owner, you're responsible for all the changes that you're trying to get merged here.

File: src/app/api/artifacts/dr-sample-cvr.ts (hunk 3 above)
Comment (awoflaherty) [reply src/app/api/artifacts/dr-sample-cvr.ts:1]: contents in the /artifacts/ directory will be removed as part of the Markit integration.  So while you can update Rosie with it, I will not recommend moving this directory to Markit.

File: package-lock.json (hunk 4 above)
Comment (awoflaherty) [reply package-lock.json:882]: @AV-MadsChris Please remove this from your PR.

File: src/app/app.component.html (hunk 2 above)
Comment (AV-MadsChris) [reply src/app/app.component.html:2]: This commit: e7b7a6c330f597ef8dc2c2412abf8f2f6d62ba71

I used a branch where you had created a `PairingCodeScreen`

File: config.xml (hunk 5 above)
Comment (AV-MadsChris) [reply config.xml:21]: This commit: e7b7a6c330f597ef8dc2c2412abf8f2f6d62ba71

I used a branch where you had created a `PairingCodeScreen`

File: package-lock.json (hunk 4 above)
Comment (AV-MadsChris) [reply package-lock.json:882]: As a note I cannot be held responsible for code you have asked me to use. Any changes shree has made I will not take responsibility for

**Comment by AV-MadsChris** [comment]
@awoflaherty We can't be expected to be responsible for changes made by shree.

File: config.xml (hunk 5 above)
Comment (mattbeattie) [reply config.xml:21]: Hey Mads, I think there's a lot of miscommunication / confusion here. I never personally added the `PairingCodeScreen`, and I'm not sure why that was added to the `proxy-updates` branch.

The timeline for the `proxy-updates` branch is particularly curious:

1. 2/23: I created the `proxy-updates` branch from latest `main` (confirmed by my Slack conversation with Sune)
2. 3/3: Sune merged 2 commits into the `proxy-updates` branch through PR #42 
3. 3/7: @shreedhanu08 pushes commit https://github.com/trustthevote/MVP-App-TestHarness/commit/e7b7a6c330f597ef8dc2c2412abf8f2f6d62ba71 (???)
4. 3/8: @shreedhanu08 pushes commit https://github.com/trustthevote/MVP-App-TestHarness/commit/adb1d9d5de648812849c5a60e0295e134497cee7 (???)
5. 3/17: PR #46 is merged, which contains 5 commits from you and Sune

The two commits pushed by @shreedhanu08 appear to have A) not come from any PR (as would be the preferred practice) and B) were not also added on the `main` branch. 

**In short, these commits just "showed up" in the `proxy-updates` branch, and we're all now seeing and reacting to them for the first time. Someone needs to tell us whether they need to be in this branch or not.** If we assume that these commits were not added by accident, that means there was a willful intent to add material changes to the `proxy-updates` branch, and that @shreedhanu08 was likely asked by someone to do so.

Mads, I missed that in my initial analysis, so thanks for bringing this up. This is definitely not a question for you.

@awoflaherty @trustthevote, can you help us with this mystery?

File: src/app/app.component.html (hunk 2 above)
Comment (mattbeattie) [reply src/app/app.component.html:2]: Let's follow up here: https://github.com/trustthevote/MVP-App-TestHarness/pull/48#discussion_r846097553

**Comment by mattbeattie** [comment]
The changes added by @AV-MadsChris look good to me. John and Anne have also expressed approval.

Unfortunately, the commits directly pushed by @shreedhanu08 to the `proxy-updates` branch - which have created an abundance of confusion for everyone involved in this process - are also preventing us from merging this PR. Here's why:

* Those commits were added to the `proxy-updates` branch directly as commits, rather than as PRs as is preferred
* Those commits included changes which broke CI, which would have been prevented if they had been added as PRs
* The `proxy-updates` branch was the basis for this branch, so these commits ended up in this PR
* The breaking changes in those commits are manifesting here as well

I don't think it should be up to Mads to fix tests for code that someone else added in a broken state. I attempted to apply a fix, but don't have the ability to commit to this branch since it lives on AV's fork.

I think the following steps are the simplest path to getting this PR merged:

1. @shreedhanu08 will need to open a PR to fix the failures in the `proxy-updates` branch and get it merged
2. @AV-MadsChris will need to pull that change into the `aion-dk:ballot-checking-integration` branch, at which point this PR should be working again

@awoflaherty @trustthevote what are your thoughts?

**Review by mattbeattie — APPROVED** [review]

File: config.xml (hunk 5 above)
Comment (mattbeattie) [reply config.xml:21]: Resolution summarized here: https://github.com/trustthevote/MVP-App-TestHarness/pull/48#issuecomment-1093634620

File: src/app/app.component.html (hunk 2 above)
Comment (mattbeattie) [reply src/app/app.component.html:2]: Resolution summarized here: https://github.com/trustthevote/MVP-App-TestHarness/pull/48#issuecomment-1093634620

**Comment by awoflaherty** [comment]
I have asked @shreedhanu08  to open a PR to fix the failures in the proxy-updates branch and get it merged.  This mya not happen until Monday.

**Comment by trustthevote** [comment]
I agree with Matt's suggestions. I'll leave to Anne (with help from Matt if she wants) to explain to Ian and team the level of dislocation and delay that the direct commits caused.

**Comment by AV-MadsChris** [comment]
I merged `proxy-updates` into my branch and skipped the failing test. However, now there seems to be a coverage issue. Is it possible for you to create a PR towards my branch with a fix for the failing test and hope that also fixes the coverage issue?

**Comment by awoflaherty** [comment]
@AV-MadsChris  - Matt is on vacation today, so let's wait until he is back at his desk and gets a chance to review the latest.

**Comment by mattbeattie** [comment]
Oh excellent, coverage enforcement IS working. 

@AV-MadsChris feel free to adjust the coverage thresholds in this file so that they're somewhere below the coverage you have on your branch. That should ensure things pass.

https://github.com/trustthevote/MVP-App-TestHarness/blob/main/karma.conf.js#L33-L38

**Comment by AV-MadsChris** [comment]
@mattbeattie @awoflaherty Looks like it's working now :-)

**Review by mattbeattie — APPROVED** [review]
Excellent - looks good to merge!
